cg/services/deliver_files/rsync/service.py

```diff
@@ -246,39 +264,85 @@ def run_rsync_for_ticket(self, ticket: str, dry_run: bool) -> int:
+            commands=commands, error_command=error_command, job_name=job_name, dry_run=dry_run
+        )
+
+    def _deliver_folder_contents(
+        self,
+        case: Case,
+        folder_creation_job_id: int,
+        folders_to_deliver: set[Path],
+        ticket: str,
+        source_and_destination_paths: dict[str, Path],
+        dry_run: bool = False,
+    ) -> int:
+        """Submit the job that rsyncs case data to the delivery server and return the job id."""
+        command: str = self.concatenate_rsync_commands(
+            folder_list=folders_to_deliver,
+            source_and_destination_paths=source_and_destination_paths,
+            ticket=ticket,
+            case=case,
+        )
+
+        return self._generate_and_submit_sbatch(
+            commands=command,
+            dry_run=dry_run,
+            error_command=ERROR_RSYNC_FUNCTION,
+            exclude=SLURM_UPLOAD_EXCLUDED_COMPUTE_NODES,
+            job_name=f"{case.internal_id}_rsync",
+            dependency=f"--dependency=afterok:{folder_creation_job_id}",
+        )
+
+    def _generate_and_submit_sbatch(
         self,
+        job_name: str,
         commands: str,
-        job_prefix: str,
-        account: str = None,
-        email: str = None,
-        log_dir: str = None,
-        hours: int = 24,
-        number_tasks: int = 1,
-        memory: int = 1,
+        error_command: str,
         dry_run: bool = False,
+        dependency: str | None = None,
+        exclude: str | None = None,
     ) -> int:
-        """Instantiates a slurm api and sbatches the given commands. Default parameters can be
-        overridden."""
         sbatch_parameters: Sbatch = Sbatch(
-            job_name="_".join([job_prefix, "rsync"]),
-            account=account or self.account,
-            number_tasks=number_tasks,
-            memory=memory,
-            log_dir=log_dir or self.log_dir.as_posix(),
-            email=email or self.mail_user,
-            hours=hours,
-            quality_of_service=self.slurm_quality_of_service,
+            account=self.account,
             commands=commands,
-            error=ERROR_RSYNC_FUNCTION.format(),
-            exclude="--exclude=gpu-compute-0-[0-1],cg-dragen",
+            dependency=dependency,
+            email=self.mail_user,
+            error=error_command,
+            exclude=exclude,
+            hours=SLURM_UPLOAD_MAX_HOURS,
+            job_name=job_name,
+            log_dir=self.log_dir.as_posix(),
+            memory=SLURM_UPLOAD_MEMORY,
+            number_tasks=SLURM_UPLOAD_TASKS,
+            quality_of_service=self.slurm_quality_of_service,
         )
+
         slurm_api = SlurmAPI()
         slurm_api.set_dry_run(dry_run=dry_run)
         sbatch_content: str = slurm_api.generate_sbatch_content(sbatch_parameters=sbatch_parameters)
-        sbatch_path = self.log_dir / "_".join([job_prefix, "rsync.sh"])
+        sbatch_path: Path = self.log_dir / f"{job_name}.sh"
```

Comment:
This is a convention that we have agreed before
```suggestion
        sbatch_path = Path(self.log_dir, f"{job_name}.sh")
```